src/main/java/mat/server/service/impl/FhirMeasureServiceImpl.java

```diff
@@ -71,31 +86,67 @@ public FhirConvertResultResponse convert(ManageMeasureSearchModel.Result sourceM
         }
 
         return fhirConvertResultResponse;
     }
 
+    /**
+     * The only purpose of this is to make it mockable.
+     * setMeasureXMLAsByteArray has Hibernate LOB crap in it that is difficult to mock away here.
+     */
+    public void saveMeasureXml(MeasureXML existingMeasureXml, String newXml) {
+        if (!TEST_MODE) {
+            existingMeasureXml.setMeasureXMLAsByteArray(newXml);
+            measureXMLDAO.save(existingMeasureXml);
+        }
+    }
 
 
-    private void persistFhirMeasure(String loggedinUserId, FhirConvertResultResponse fhirConvertResultResponse, ManageMeasureDetailModel sourceMeasureDetails, Optional<String> fhirCqlOpt) {
+    private void persistFhirMeasure(String loggedinUserId,
+                                    FhirConvertResultResponse fhirConvertResultResponse,
+                                    ManageMeasureDetailModel sourceMeasureDetails,
+                                    String convertedCql) {
         // Just to make sure the change is atomic and performed within the same single transaction.
         transactionTemplate.executeWithoutResult(status -> {
             try {
                 ManageMeasureSearchModel.Result fhirMeasure = measureCloningService.cloneForFhir(sourceMeasureDetails);
                 fhirConvertResultResponse.setFhirMeasure(fhirMeasure);
-                SaveUpdateCQLResult cqlResult = measureLibraryService.saveCQLFile(fhirMeasure.getId(), fhirCqlOpt.get());
-                if (!cqlResult.isSuccess()) {
-                    throw new MatRuntimeException("Mat cannot persist converted FHIR measure CQL file.");
-                }
+
+                //Update the MAT xml.
+                convertXml(fhirMeasure.getId(), convertedCql);
+
                 measureLibraryService.recordRecentMeasureActivity(fhirMeasure.getId(), loggedinUserId);
-            } catch (MatException e) {
-                throw new MatRuntimeException(e);
+            } catch (MatException | MatRuntimeException e) {
+                logger.error("persistFhirMeasure error",e);
+                throw new MatRuntimeException("Mat cannot persist converted FHIR measure CQL file.");
             }
         });
     }
 
+    private void convertXml(String measureId, String cql) throws MatException {
+        try {
+            MeasureXML measureXml = measureXMLDAO.findForMeasure(measureId);
+            String sourceMeasureXml = measureXml.getMeasureXMLAsString();
+
+            XmlProcessor processor = new XmlProcessor(sourceMeasureXml);
+            String cqlModelXmlFrag = processor.getXmlByTagName("cqlLookUp");
+            CQLModel sourceCqlModel = (CQLModel) xmlMarshalUtil.convertXMLToObject("CQLModelMapping.xml",
+                    cqlModelXmlFrag,
+                    CQLModel.class);
+
+            CqlToMatXml converter = new CqlToMatXml(sourceCqlModel, cql);
+            CQLModel destinationCqlModel = converter.convert();
+
```

Comment:
It looks much easier and simpler than the original code. But the original saveCQLFile file has a lot of logic to cleanup valuesets/populations/codes and additional xml updates. 
Do you believe it's not needed in case of conversion?

Reply:
It was causing too many problems. The microservice cql is pretty clean and the previous cql was already packaged.